File: Ywy/JavaScript/level2/test/ParenthesesConversion.test.js
import ParenthesesConversion from '../ParenthesesConversion';

test(`"(()())()" is "(()())()"`, () => {
    expect(new ParenthesesConversion().solution("(()())()")).toBe("(()())()");
});

test(`"()))((()" is "()(())()"`, () => {
    expect(new ParenthesesConversion().solution("()))((()")).toBe("()(())()");
});

File: Ywy/JAVA/src/level2/test/cases/ParenthesesConversionTest.java
package level2.test.cases;

import static org.junit.Assert.assertEquals;

import org.junit.Before;
import org.junit.Test;

import level2.ParenthesesConversion;

public class ParenthesesConversionTest {
	private String[] ps;
	private String[] answers;
	private ParenthesesConversion parenthesesConversion;
	@Before
	public void setUp() throws Exception {
		ps = new String[] {
			")()()()(",
			"()))((()",
			"(()())()",
			")(",
			"()",
			"())(",
		};
		
		answers = new String[] {
			"(((())))",
			"()(())()",
			"(()())()",
			"()",
			"()",
			"()()",
		};
		parenthesesConversion = new ParenthesesConversion();
	}

	@Test
	public void test() {
		int index = 0;
		int range = ps.length;
		do {
			assertEquals(answers[index], parenthesesConversion.solution(ps[index]));
		} while(++index < range);
	}

}
